testtools/matchers/_warnings.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2009-2016 testtools developers. See LICENSE for details.
+
+__all__ = [
+    'Warnings',
+    'warning_message',
+    'is_deprecated']
+
+import warnings
+
+from ._basic import Is
+from ._const import Always
+from ._datastructures import MatchesListwise, MatchesStructure
+from ._higherorder import (
+    AfterPreprocessing,
+    Annotate,
+    MatchesAll,
+    Not,
+    )
+from ._impl import Mismatch
+
+
+def warning_message(category_type, message=None, filename=None, lineno=None,
+                    line=None):
+    """
+    Create a matcher that will match `warnings.WarningMessage`\s.
+
+    :param category_type: A warning type.
+    :param message_matcher: Match the warning message against this.
+    :param filename_matcher: Match the warning filename against this.
+    :param lineno_matcher: Match the warning line number against this.
+    :param line_matcher: Match the warning line of code against this.
+    """
+    category_matcher = Is(category_type)
+    message_matcher = message or Always()
+    filename_matcher = filename or Always()
+    lineno_matcher = lineno or Always()
+    line_matcher = line or Always()
+    return MatchesStructure(
+        category=Annotate(
+            "Warning's category type does not match",
+            category_matcher),
+        message=Annotate(
+            "Warning's message does not match",
+            AfterPreprocessing(str, message_matcher)),
+        filename=Annotate(
+            "Warning's filname does not match",
+            filename_matcher),
+        lineno=Annotate(
+            "Warning's line number does not match",
+            lineno_matcher),
+        line=Annotate(
+            "Warning's source line does not match",
+            line_matcher))
+
+
+class Warnings(object):
+    """
+    Match if the matchee produces deprecation warnings.
```

Comment:
I suggest removing the word 'deprecation' - the warning could be about anything, not just deprecated code, right?